starter_kits/C/src/MyBot.c

```diff
@@ -0,0 +1,324 @@
+	if(scanf("{\"CAPTURE_ENABLED\":%*5[^,],\"CAPTURE_RADIUS\":%*u,\"DEFAULT_MAP_HEIGHT\":%*u,\"DEFAULT_MAP_WIDTH\":%*u,\"DROPOFF_COST\":%hu,\"DROPOFF_PENALTY_RATIO\":%*u,\"EXTRACT_RATIO\":%hhu,\"FACTOR_EXP_1\":%*f,\"FACTOR_EXP_2\":%*f,\"INITIAL_ENERGY\":%*u,\"INSPIRATION_ENABLED\":%5[^,],\"INSPIRATION_RADIUS\":%hhu,\"INSPIRATION_SHIP_COUNT\":%hhu,\"INSPIRED_BONUS_MULTIPLIER\":%hhu.0,\"INSPIRED_EXTRACT_RATIO\":%hhu,\"INSPIRED_MOVE_COST_RATIO\":%hhu,\"MAX_CELL_PRODUCTION\":%*u,\"MAX_ENERGY\":%hu,\"MAX_PLAYERS\":%*u,\"MAX_TURNS\":%hu,\"MAX_TURN_THRESHOLD\":%*u,\"MIN_CELL_PRODUCTION\":%*u,\"MIN_TURNS\":%*u,\"MIN_TURN_THRESHOLD\":%*u,\"MOVE_COST_RATIO\":%hhu,\"NEW_ENTITY_ENERGY_COST\":%hu,\"PERSISTENCE\":%*f,\"SHIPS_ABOVE_FOR_CAPTURE\":%*u,\"STRICT_ERRORS\":%*5[^,],\"game_seed\":%*u}", &dropoff_cost, &extract_ratio, (char *)&inspiration_enabled_str, &inspiration_radius, &inspiration_ship_count, &inspired_bonus_multiplier, &inspired_extract_ratio, &inspired_move_cost_ratio, &max_energy, &max_turns, &move_cost_ratio, &ship_cost) < 12){
+		exit(1);
+	}
+
+	if(inspiration_enabled_str[0] == 't')
+		inspiration_enabled = 1;
+	else
+		inspiration_enabled = 0;
+
+
+	// Get number of players and player ID
+	if(scanf("%hhu %hhu", &number_of_players, &my_id) < 1){
+		exit(1);
+	}
+
+	// Now that we have our player ID, we add it to time(NULL) to seed our "smart" RNG
+	srand(time(NULL) + my_id);
+
+	// Get each player's ship yard point coordinates
+	for(i = 0; i < number_of_players; i++){
+		if(scanf("%hhu %hhu %hhu", &sca, &scb, &scc) < 3){
+			exit(1);
+		}
+
+		players[sca].ship_yard[0] = scb;
+		players[sca].ship_yard[1] = scc;
+	}
+
+	// Read the map size
+	if(scanf("%hhu %hhu", &map_size[0], &map_size[1]) < 2){
+		exit(1);
+	}
+
+	// Now that we have the map size, let's set the map half width and map half height variables
+	mhw = map_size[0] >> 1;
+	mhh = map_size[1] >> 1;
+
+	// Finally, we read the initial map
+	for(i = 0; i < map_size[0]; i++){
+		for(j = 0; j < map_size[1]; j++){
+			if(scanf("%hu", &map[j][i].halite) < 1){
+				exit(1);
+			}
+		}
+	}
+
+	// Now that we've read the starting data, it's time to output our bot's name
+	printf("MyBot\n");
+	// stdout has to be flushed before the game engine will actually get the commands
+	fflush(stdout);
+
+
+	// Now we start the game loop
+	while(1){
+		// Before we deal with the game engine, we should reset whether or not each map tile is occupied
+		for(i = 0; i < map_size[1]; i++){
+			for(j = 0; j < map_size[0]; j++){
+				map[j][i].occupied = (unsigned short)-1;
+			}
+		}
+
+
+		// Read the turn number
+		if(scanf("%hu", &turn_number) < 1){
+			exit(1);
+		}
+
+		// Now we're going to save the turn number to the log file just to show how it's done
+		fprintf(log_file, "Starting turn %hu\n", turn_number);
+		fflush(log_file);
+
+		// Read information about how each of the players are doing
+		for(i = 0; i < number_of_players; i++){
```

Comment:
This loop should be a function

Reply:
Pretty much all the below code blocks should be functions too.

Reply:
As much as I despise the idea, I'm open to making a bunch of functions, but what is the advantage to putting into functions piece of code that are fairly short and only appear in one place in the program? If your argument is readability, isn't that what comments are for?